keps/sig-node/4817-resource-claim-device-status/README.md

```diff
@@ -0,0 +1,618 @@
+them.
+
+#### Story 2 - Network Device Status for Troubleshooting
+
+As a Network Administrator, troubleshooting networking issues can be complex
+and time consuming especially when the device characteristics and operational
+status are not readily accessible. The `DeviceStatuses` field in the
+`ResourceClaim.Status` provides access to comprehensive details regarding
+network interfaces helping to quickly and efficiently identify the issues such
+as error messages on failed network interface configuration, incorrect IP
+assignments or misconfigured network interfaces.
+
+### Notes/Constraints/Caveats (Optional)
+
+The content of `DeviceInfo` is driver specific and not standardized as part of
+DRA, the interpretation of this field may then vary between controllers and
+users reading it. 
+
+The accuracy of the information depends on the implementation of the DRA
+Drivers, the reported status of the device may not always reflect the real time
+changes of the state of the device.
+
+### Risks and Mitigations
+
+As stated, 3rd party DRA drivers will set and update the `DeviceStatuses` for
+the device they manage. An access control must be set in place to restrict the
+write access to the appropriate driver (A device status can only be updated by
+the driver which allocated and configured this device).
+
+Adding `DeviceInfo` as an arbitrary data slice may introduce extra processing
+and storage overhead which might impact performance in a cluster with many
+devices and frequent status updates. In large-scale clusters where many devices
+are allocated, this impact must be considered.
+
+## Design Details
+
+### API
+
+The `ResourceClaimStatus` struct in `pkg/apis/resource/types.go` will be
+extended to include the slice of `DeviceStatuses`.
+
+`ResourceClaim` validation of the status in
+`pkg/apis/resource/validation/validation.go` will be covered to allow a device
+to be reported only once in the slice, a device is being identified  by
+`<driver name>/<pool name>/<device name>`.
+
+### Test Plan
+
+[x] I/we understand the owners of the involved components may require updates to
+existing tests to make this code solid enough prior to committing the changes necessary
+to implement this enhancement.
+
+##### Prerequisite testing updates
+
+##### Unit tests
+
+- `ResourceClaim` validation:
+    - A device can only be reported once in the `ResourceClaim`.
+    - The reported device is allocated in the `ResourceClaim`.
+    - Properties set in `AllocatedDeviceStatus` are in the correct format.
+
+##### Integration tests
+
+- Usage of the `DeviceStatuses` field in the `ResourceClaimStatus`:
+    * With the feature gate enabled, the field exists in the `ResourceClaim`.
+    * With the feature gate disabled, the field does not exist in the
+      `ResourceClaim`.
+
+##### e2e tests
+
+TBD
+
+### Graduation Criteria
+
+#### Alpha
+
+- Feature implemented behind feature gates (`ResourceClaimDeviceStatus`).
```

Comment:
do we want to start a new track for adding a new field? is not possible to reuse existing track @johnbelamaric @pohly @klueska

Reply:
I don't think it fits into some other track. It's a new feature with new fields specifically for it.

Reply:
Do we want to start DRA-related feature gates with "DRA"? I did for `DRAControlPlaneController` (= classic DRA).

Reply:
ack

Reply:
Yeah, we can't include it in another feature gate and still move forward independently.

Reply:
I renamed it as `DRAControlPlaneController` as suggested.

Reply:
I wasn't suggesting to use exactly that - it's already taken! :sweat_smile: 

Perhaps use `DRAClaimDeviceStatus`?

Reply:
Ho yes, sorry, I wanted to say `DRAResourceClaimDeviceStatus`, not `DRAControlPlaneController`